File: Agent/Globals/Classes/Analysis/AutoAnalysisDeckFields.py
class AutoAnalysisDeckFields:
    """
    Namespaced keys that the auto-analysis subsystem stores under
    deck.additionalData. Mirrors the JS counterpart in
    Main/Globals/Classes/Analysis/AutoAnalysisDeckFields.js. The values
    are the literal property names that Mongo and the sync layer see.

    Previously these constants were inlined as class statics on
    AnalyzeDeckPerformance; they were extracted here so the curated
    pipeline (analysis, generation, controller) can share one source of
    truth.
    """

    AUTO_PERFORMANCE_ANALYSIS_ENABLED = "autoPerformanceAnalysisEnabled"
    AUTO_GENERATE_CURATED_STUDY_ENABLED = "autoGenerateCuratedStudyEnabled"
    LAST_ANALYZED_AT = "lastAnalyzedAt"
    LAST_ANALYSIS_TOPICS = "lastAnalysisTopics"
    LAST_CURATED_BATCH_TAG = "lastCuratedBatchTag"
    LAST_CURATED_BATCH_TOPICS = "lastCuratedBatchTopics"
    LAST_SKIPPED_DUE_TO_IN_PROGRESS_AT = "lastSkippedDueToInProgressAt"

    @staticmethod
    def get_all_keys():
        return [
            AutoAnalysisDeckFields.AUTO_PERFORMANCE_ANALYSIS_ENABLED,
            AutoAnalysisDeckFields.AUTO_GENERATE_CURATED_STUDY_ENABLED,
            AutoAnalysisDeckFields.LAST_ANALYZED_AT,
            AutoAnalysisDeckFields.LAST_ANALYSIS_TOPICS,
            AutoAnalysisDeckFields.LAST_CURATED_BATCH_TAG,
            AutoAnalysisDeckFields.LAST_CURATED_BATCH_TOPICS,
            AutoAnalysisDeckFields.LAST_SKIPPED_DUE_TO_IN_PROGRESS_AT,
        ]

File: Dock/Globals/Classes/Analysis/AutoAnalysisDeckFields.js
/**
 * Namespaced keys that the auto-analysis subsystem stores under
 * `deck.additionalData`. Mirrors the canonical definition in
 * Main/Globals/Classes/Analysis/AutoAnalysisDeckFields.js. Dock keeps a
 * local copy so server-side query engines never sprinkle string
 * literals when filtering by analysis state. The values themselves are
 * the literal property names that Mongo and the sync layer see.
 */
class AutoAnalysisDeckFields
{
    static AUTO_PERFORMANCE_ANALYSIS_ENABLED = "autoPerformanceAnalysisEnabled";
    static AUTO_GENERATE_CURATED_STUDY_ENABLED = "autoGenerateCuratedStudyEnabled";
    static LAST_ANALYZED_AT = "lastAnalyzedAt";
    static LAST_ANALYSIS_TOPICS = "lastAnalysisTopics";
    static LAST_CURATED_BATCH_TAG = "lastCuratedBatchTag";
    static LAST_CURATED_BATCH_TOPICS = "lastCuratedBatchTopics";
    static LAST_SKIPPED_DUE_TO_IN_PROGRESS_AT = "lastSkippedDueToInProgressAt";

    static getAllKeys()
    {
        return [
            AutoAnalysisDeckFields.AUTO_PERFORMANCE_ANALYSIS_ENABLED,
            AutoAnalysisDeckFields.AUTO_GENERATE_CURATED_STUDY_ENABLED,
            AutoAnalysisDeckFields.LAST_ANALYZED_AT,
            AutoAnalysisDeckFields.LAST_ANALYSIS_TOPICS,
            AutoAnalysisDeckFields.LAST_CURATED_BATCH_TAG,
            AutoAnalysisDeckFields.LAST_CURATED_BATCH_TOPICS,
            AutoAnalysisDeckFields.LAST_SKIPPED_DUE_TO_IN_PROGRESS_AT,
        ];
    }
}

module.exports = AutoAnalysisDeckFields;
